### PR TITLE
Addressing Issue #3691 - SplitButton: In Safari, the SplitButton has mismatched heights

### LINKED ISSUE #3691: SplitButton: In Safari, the SplitButton has mismatched heights
### Bug Report

- __Package version(s)__: (fill this out) 5.29
- __Browser and OS versions__: (fill this out if relevant) Safari 11

Are you willing to submit a PR to fix? (Yes, No) Yes 

Requested priority: (Blocking, High, Normal, Low) Normal

#### Describe the issue:

In Safari, the split button does not render with the correct height

screen shot 2018-01-09 at 12 46 45 pm

### PR DESCRIPTION
**Pull request by oengusmacinog**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #3691 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Added a margin: 0 reset to **ms-Button** root, as well as the split button.  This is a fix for Safari, where the browser default seems to set the margin to `0em` which for some reason translates to a margin left and right of 2px throwing off the button layout.

I will not this bug does not show up on the live demo site ([https://developer.microsoft.com/en-us/fabric#/components/button](https://developer.microsoft.com/en-us/fabric#/components/button)) 
because the UHF applies its own custom file with resets (looks, like normalize, is part of it) with this file: **mwf-west-european-default.min.css**.

I will follow up on general resets for Fabric as well, but for now, this single reset will solve the problem with the button.

### CODE REVIEW

**Comment by micahgodbolt** [comment]
i'd like to understand this one a bit better. it seems really odd behavior. The problem with adding zero margins is that other's might have set margins that will now see them overridden. Is this a safari bug that will be resolved later?

**Comment by micahgodbolt** [comment]
I looked into this bug a bit more and wow, thats annoying! This happened with overflow as well (different default in IE11) and the solution was to add it to the Fabric component. My problem with this solution is that it has a chance of breaking any margins applied with a single classname, but if we add it to the button tag itself, any class won't be affected. 

image

**Comment by oengusmacinog** [comment]
I like it! :)

**Comment by micahgodbolt** [comment]
well that sure cleans up the PR a bit :D

File: packages/experiments/src/components/CommandBar/__snapshots__/CommandBar.test.tsx.snap [hunk 1]
```diff
@@ -76,6 +76,10 @@ exports[`CommandBar renders commands correctly 1`] = `
                   font-size: 14px;
                   font-weight: 400;
                   height: 100%;
+                  margin-bottom: 0px;
```
Comment (micahgodbolt) [inline packages/experiments/src/components/CommandBar/__snapshots__/CommandBar.test.tsx.snap]: why are these still here?

**Comment by JasonGore** [comment]
It doesn't look like the experiments change file is needed anymore?

**Comment by oengusmacinog** [comment]
Currently running `rush rebuild` to clean up that experiments snapshot as well, but yessss way cleaner  :)

**Pull request merged**